### DISCUSSION TITLE
Version roadmap consultation

### DISCUSSION THREAD

**Discussion by daohu527** [eProsima/Fast-DDS#5090] [Q&A] [answered]
### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

Since bazel now supports code [repositories](https://registry.bazel.build/search?module=foonathan_memory&q=fastcdr), I want to support fastdds directly in the official website library, and it has been compiled based on the latest master implementation

https://github.com/daohu527/Fast-DDS/tree/master

## question

So I'm eager to know our release schedule so that I can release the first bazel version, I found that there are 2 branches 2.14.x and master both have updates, and the src/cpp/CMakeLists.txt are very different.

Can you give me some advice?

### Current behavior

no

### Steps to reproduce

no

### Fast DDS version/commit

no

### Platform/Architecture

Ubuntu Focal 20.04 amd64

### Transport layer

Default configuration, UDPv4 & SHM

**Comment by JesusPoderoso** [comment]
Hi @daohu527, thanks for using Fast DDS.
Sorry for the inconvenience. As you have read in our repository's [README](https://github.com/eProsima/Fast-DDS), we are close to the v3.0.0 release. Fast DDS' master branch is undergoing major changes entailing API breaks. Until Fast DDS v3.0.0 is released, and it is strongly advisable to use the latest stable branch, [2.14.x](https://github.com/eProsima/Fast-DDS/tree/2.14.x). That is the reason why the CMakeLists.txt files are so different.

In the [release support](https://github.com/eProsima/Fast-DDS/blob/master/RELEASE_SUPPORT.md) section, there is more information about the currently supported versions, their lifecycle and the policy for backporting ABI compatible features and bugfixes.
I hope this clarifies your doubts.

Also, I am moving this issue to the proper Q&A forum according to the [Fast DDS CONTRIBUTING guidelines](https://github.com/eProsima/Fast-DDS/blob/master/CONTRIBUTING.md).

**Comment by daohu527** [comment]
@JesusPoderoso Thanks for your reply. When will 3.0 be released? I am now faced with a choice: either wait for 3.0 or adapt 2.14.x first and then release it to the bazel repository.

**Reply by JesusPoderoso** [reply, marked as answer]
Hi @daohu527, Fast DDS v3.0.0 has just been released!

**Reply by daohu527** [reply]
Does it mean that we can use version 3.0 in the future? Is version 2.0 still maintained? Which version is recommended?

**Reply by JesusPoderoso** [reply]
Yes, check each release lifecycle in the [release support](https://github.com/eProsima/Fast-DDS/blob/master/RELEASE_SUPPORT.md) section.
The recommended version depends on your needs. New v3.0.0 is XTypes compliant, but 2.14.x has been more tested and it is LTS.

**Reply by daohu527** [reply]
Thank you for your patient explanation. I am currently building a new autonomous driving system [wheelos](https://github.com/wheelos/core) based on fastdds. I want it to be as efficient as possible and support new features, such as object-based subscriptions. 

I hope fastdds will become more and more powerful

**Reply by JesusPoderoso** [reply]
Looking forward to seeing more about the project!
